workers/miscale.py

```diff
@@ -16,14 +19,48 @@
 class MiscaleWorker(BaseWorker):
 
     SCAN_TIMEOUT = 5
-
+    
+    def getAge(self, d1):
+        d1 = datetime.strptime(str(d1), "%Y-%m-%d")
+        d2 = datetime.strptime(datetime.today().strftime('%Y-%m-%d'), '%Y-%m-%d')
+        return abs((d2 - d1).days)/365
+    
     def status_update(self):
         results = self._get_data()
+        
         messages = [MqttMessage(topic=self.format_topic("weight/" + results.unit), payload=results.weight)]
         if results.impedance:
             messages.append(MqttMessage(topic=self.format_topic("impedance"), payload=results.impedance))
         if results.midatetime:
             messages.append(MqttMessage(topic=self.format_topic("midatetime"), payload=results.midatetime))
+        
+        if hasattr(self, 'users'):
+            for key, item in self.users.items():
+                if results.weight > item['weight_template']['min'] and results.weight < item['weight_template']['max']:
+                    user = key
+                    sex = item['sex']
+                    height = item['height']
+                    age = self.getAge(item['dob'])
+                    
+                    lib = bodyMetrics(results.weight, height, age, sex, 0)
+                    metrics = {
+                        'weight': float("{:.2f}".format(results.weight)),
+                        'bmi': float("{:.2f}".format(lib.getBMI())),
+                        'basal_metabolism': float("{:.2f}".format(lib.getBMR())),
+                        'visceral_fat': float("{:.2f}".format(lib.getVisceralFat()))
+                    }
+                    
+                    if results.impedance:
+                        lib = bodyMetrics(results.weight, height, age, sex, int(results.impedance))
+                        metrics['lean_body_mass'] = float("{:.2f}".format(lib.getLBMCoefficient()))
+                        metrics['body_fat'] = float("{:.2f}".format(lib.getFatPercentage()))
+                        metrics['water'] = float("{:.2f}".format(lib.getWaterPercentage()))
+                        metrics['bone_mass'] = float("{:.2f}".format(lib.getBoneMass()))
+                        metrics['muscle_mass'] = float("{:.2f}".format(lib.getMuscleMass()))
+                        metrics['protein'] = float("{:.2f}".format(lib.getProteinPercentage()))
+                    
+                    metrics['timestamp'] = results.midatetime
+                    messages.append(MqttMessage(topic=self.format_topic("users/" + user), payload=json.dumps(metrics)))
```

Comment:
I think this can be removed, mqtt lib should handle serialization on its own. Just pass metrics here